Added details after processing feedback for BE (#544)
valtimo-platform/valtimo-documentation: release-notes/major12/12.0.0/valtimo-backend-libraries.md

  Additionally, `CamundaProcessJsonSchemaDocumentService.getDocument(DelegateExecution execution)` has been removed.
  This method is replaced by `DocumentDelegateService.getDocument(DelegateExecution execution`.

* **ResourceService implementation is now optional**
  Valtimo can now start without providing any implementation of the `ResourceService`.

>   However, when no implementation is provided, the following features will not work:

Reviewer: ```suggestion
  When no implementation is provided, the following features will not work:
```

  - The `camundaSmartDocumentGenerator` and `smartDocumentGenerator` beans will not be available. The plugin should work as normal.
  - The `documentRelatedFileSubmittedEventListenerImpl` bean will not be available.
  - `JsonSchemaDocumentService.assignResource` will throw an error when invoked.
  - The `FormIoFormFileResource` bean will not be available.

* **`OpenZaakUrlProvider` has been replaced

>   The `OpenZaakUrlProvider` class and bean (`openZaakUrlProvider`) has been removed. 

Reviewer: ```suggestion
  The `OpenZaakUrlProvider` class and bean (`openZaakUrlProvider`) have been removed. 
```

>   It has been replaced by:

Reviewer: ```suggestion
  These have been replaced by:
```

>   - `ZaakUrlProvider`: implemented by `DefaultZaakUrlProvider` in the zaken-api module

Reviewer: ```suggestion
  - `ZaakUrlProvider`: implemented by `DefaultZaakUrlProvider` in the zaken-api module.
```

  - `ZaaktypeUrlProvider`: implemented by `DefaultZaaktypeUrlProvider` in the zaken-api module

> * **`ZaakUrlProvider.getZaak(documentId: UUID): String` has been removed

Reviewer: ```suggestion
* **`ZaakUrlProvider.getZaak(documentId: UUID): String` has been removed.
```

>   Replaced by `ZaakUrlProvider.getZaakUrl(documentId: UUID): URI`

Reviewer: ```suggestion
  This method is replaced by `ZaakUrlProvider.getZaakUrl(documentId: UUID): URI`.
```

Instructions on how to migrate to this version of Valtimo can be found [here](migration.md).

* **OpenZaak module**

>   The OpenZaak module has been deprecated. The deprecated methods have been annotated with instructions on how to replace them.

Reviewer: ```suggestion
  The OpenZaak module has been deprecated. The deprecated methods and classes have been annotated with instructions on how to replace them.
```

>   In general: please use the available (Plugin) functionality from the ZWG modules.

Reviewer: ```suggestion
  The general guidelines here are use the available (Plugin) functionality from the ZWG modules as a replacement.
```
